contracts/Example2.sol

```diff
@@ -3,9 +3,13 @@ pragma solidity ^0.8.4;
 contract Example2 {
     uint public counter;
  
-    function incrementBy(uint[] memory arr) public {
-        for (uint idx = 0; idx < arr.length; idx++){
-            counter += arr[idx];
+    function incrementBy(uint[] calldata arr) external {
+        uint _counter;
+        uint[] calldata _arr = arr;
+        for (uint idx = 0; idx < _arr.length; idx++){
```

Comment:
`_arr.length` can be stored in a separate variable to avoid calculating it multiple times